ledger/participant-integration-api/src/test/lib/scala/platform/store/backend/StorageBackendTestsUserManagement.scala

```diff
@@ -96,22 +96,68 @@ private[backend] trait StorageBackendTestsUserManagement
     getNonexistent shouldBe None
   }
 
-  it should "get users (getUsers)" in {
-    val user1 = newUniqueUser()
-    val user2 = newUniqueUser()
-    val emptyUsers = executeSql(tested.getUsers())
+  it should "get all users (getUsers) ordered by id" in {
+    val user1 = newUniqueUser(userIdPrefix = "user_id_1_")
```

Comment:
Why do you need to specify the user id prefix? If it is to make the test results easier to read, then why not hardcode the whole user name (i.e., remove the UUID suffix)? Each test starts with an empty database, so there won't be any collisions with user ids from other tests.

Reply:
I used hand-picked prefixes because assertions in this test case depend on the order of returned elements.
This might be confusing so changing `userIdPrefix` parameter to `userId` and the underlying impl to:

```
  private def newUniqueUser(
      emptyPrimaryParty: Boolean = false,
      userId: String = "",
  ): User = {
    val uuid = UUID.randomUUID.toString
    val primaryParty =
      if (emptyPrimaryParty)
        None
      else
        Some(Ref.Party.assertFromString(s"primary_party_${uuid}"))
    val userIdStr = if(userId != "") userId else s"user_id_${uuid}"
    User(
      id = Ref.UserId.assertFromString(userIdStr),
      primaryParty = primaryParty,
    )
  }
```